internal/dao/stubs/pubkey_dao.go

```diff
@@ -19,36 +19,49 @@ func init() {
 	dao.GetPubkeyDao = getPubkeyDao
 }
 
-type ContextReadError struct{}
+type contextReadError struct{}
 
-func (m *ContextReadError) Error() string {
+func (m *contextReadError) Error() string {
 	return "failed to find or convert dao stored in testing context"
 }
 
 func WithPubkeyDao(parent context.Context, init_store []*models.Pubkey) context.Context {
-	ctx := context.WithValue(parent, pubkeyCtxKey, PubkeyDaoStub{init_store})
+	ctx := context.WithValue(parent, pubkeyCtxKey, &PubkeyDaoStub{init_store})
 	return ctx
 }
 
-func getPubkeyDao(ctx context.Context) (dao.PubkeyDao, error) {
-	var err error
-	pkdao, ok := ctx.Value(pubkeyCtxKey).(dao.PubkeyDao)
-	if !ok {
-		err = &ContextReadError{}
+func PubkeyStubCount(ctx context.Context) (int, error) {
+	pkdao, err := getPubkeyDaoStub(ctx)
+	if err != nil {
+		return 0, err
 	}
+	return len(pkdao.store), nil
+}
+
+func getPubkeyDao(ctx context.Context) (dao.PubkeyDao, error) {
+	return getPubkeyDaoStub(ctx)
+}
 
+func getPubkeyDaoStub(ctx context.Context) (pkdao *PubkeyDaoStub, err error) {
+	var ok bool
+	if pkdao, ok = ctx.Value(pubkeyCtxKey).(*PubkeyDaoStub); !ok {
+		err = &contextReadError{}
+	}
 	return pkdao, err
 }
 
-func (PubkeyDaoStub) Create(ctx context.Context, pk *models.Pubkey) error { return nil }
-func (PubkeyDaoStub) CreateWithResource(ctx context.Context, pk *models.Pubkey, pkr *models.PubkeyResource) error {
+func (mock *PubkeyDaoStub) Create(ctx context.Context, pk *models.Pubkey) error {
+	mock.store = append(mock.store, pk)
```

Comment:
This is good for now, in the future we might develop some kind of "Storer" interface for fetching, finding and deleting mocks.